fundingmanager.go

```diff
@@ -824,17 +779,94 @@ func (f *fundingManager) reservationCoordinator() {
 	}
 }
 
-// advanceFundingState will advance the channel fromt the markedOpen state to
-// the point where the channel is ready for operation. This includes sending
-// funding locked to the peer, adding the channel to the router graph, and
-// announcing the channel. The updateChan can be set non-nil to get
-// OpenStatusUpdates.
+// advanceFundingState will advance the channel through the steps after the
+// funding transaction is broadcasted, up until the point where the channel is
+// ready for operation. This includes waiting for the funding transaction to
+// confirm, sending funding locked to the peer, adding the channel to the
+// router graph, and announcing the channel. The updateChan can be set non-nil
+// to get OpenStatusUpdates.
 //
 // NOTE: This MUST be run as a goroutine.
 func (f *fundingManager) advanceFundingState(channel *channeldb.OpenChannel,
-	updateChan chan<- *lnrpc.OpenStatusUpdate) {
+	pendingChanID [32]byte, updateChan chan<- *lnrpc.OpenStatusUpdate) {
 	defer f.wg.Done()
 
+	// If the channel is still pending we must wait for the funding
+	// transaction to confirm.
+	if channel.IsPending {
+		shortChanID, err := f.waitForFundingWithTimeout(channel)
```

Comment:
To reduce the vertical distance a bit in this method, this should be tucked away into another method IMO. It would then make it easier to test isolation as well.

Reply:
Done: https://github.com/lightningnetwork/lnd/pull/1910/commits/3e9e509128dcd9b919f24f363d58f85661e8de30